xwiki-platform-core/xwiki-platform-user/xwiki-platform-user-script/src/main/java/org/xwiki/user/script/GroupScriptService.java

```diff
@@ -140,4 +140,29 @@
     {
         return getMembers(group, true);
     }
+
+    /**
+     * Checks if a group can be added to the members of a group.
+     *
+     * @param target The targeted group
+     * @param member The group to be added
+     * @return {@code true} if the candidate member can be added to the members of the target, {@code false} otherwise.
+     * @since 12.8RC1
+     */
+    public boolean canAddAsMember(DocumentReference target, DocumentReference member) throws GroupException
+    {
+        final boolean ret;
+        if (target == null || member == null || member.equals(target)) {
+            ret = false;
+        } else {
+            // check if the candidate member is not already one of the members of the target
+            if (getGroupsFromAllWikis(target).contains(member)) {
+                ret = false;
+            } else {
+                // check if the target is not a member of the member
+                ret = !getMembers(target, true).contains(member);
```

Comment:
I don't really understand why you are using both `getGroupsFromAllWikis` and `getMembers`. I would use `getMembers` for both checks since it's generally much faster (especially in an instances with many wikis).

Reply:
I agree, I'll update to only use getMembers.